### PR TITLE
Build C libraries with the Python build system

### LINKED ISSUE #713: Build C libraries with the Python build system
Currently numerous C libraries (e.g. lz4, libxml) and also some packages (e.g. six, parso)  are packaged with Makefiles in the root folder of pyodide. I this can be confusing for new contributors and is not sustainable long term, as it makes it hard to distinguish what is pyodide code and what are packages.

Related to https://github.com/iodide-project/pyodide/issues/38

I think we should,
 - [x] move all of those libraires under `packages/` folder (while keeping the use of Makefiles). This would be just a path change. Done in https://github.com/iodide-project/pyodide/pull/714
 - [ ] migrate these libraries to be defined with a `meta.yaml` and build them with the Python build system. This would allow managing dependencies better (e.g. build LAPACK only when scipy is built etc).

### PR DESCRIPTION
**Pull request by dalcde**
Fixes #713

### CODE REVIEW

File: packages/CLAPACK/meta.yaml [hunk 1]
```diff
@@ -1,5 +1,38 @@
...
+    # In CLAPACK's Makefiles, some commands are mistakenly (?) hardcoded
+    # instead of using the right variables
+    sed -i 's/^	-ranlib /^	$(RANLIB)/' **/Makefile
+    sed -i 's/^	ar /^	$(ARCH)/' **/Makefile
+    sed -i 's/^	ld /^	$(LD)/' **/Makefile
+
+    # We build BLAS/LAPACK only for target.
+    # On host we include -LCLAPACK-3.2.1 path which has no effect on host.
+    # On target it gets rewritten by pywasmcross to the full patch of
+    # blas_WA.bc, lapack_WA.bc which are linked statically in scipy
```
Comment (rth) [inline packages/CLAPACK/meta.yaml:36]:
```suggestion
    # blas_WA.a, lapack_WA.a which are linked statically in scipy
```
with latest commits

File: pyodide_build/buildpkg.py [hunk 2]
```diff
@@ -92,7 +92,11 @@ def download_and_extract(
                 tarballname = tarballname[: -len(extension)]
                 break
 
-        return buildpath / tarballname
+        if "extract_dir" in pkg["source"]:
+            return buildpath / pkg["source"]["extract_dir"]
+        else:
+            return buildpath / tarballname
```
Comment (rth) [inline pyodide_build/buildpkg.py:98]:
```suggestion
        return buildpath / pkg.get("source", {}).get("extract_dir", tarballname)
```

**Review by rth — COMMENTED** [review]
The changes generally look good, however there is a CI issue. I'm not sure if it's related to this PR or not.

**Comment by dalcde** [comment]
There was both a CI issue and a genuine problem. CI seems to be a lot more flaky lately, which may or may not be due to changes on our part

**Review by rth — APPROVED** [review]

**Comment by rth** [comment]
@dalcde Can you add me as a contributor to your fork so that I can restart CI? Failures in scipy on Chrome are suspicious, but they also happened https://github.com/iodide-project/pyodide/pull/987, so it might be unrelated, and CI on master was green..

**Pull request merged**